src/TextInput.js

```diff
@@ -1,167 +1,154 @@
+    if (email) {
+      id = `email${idgen()}`;
     }
+    return id;
+  };
+
+  const [newValue, setNewValue] = useState(value);
+
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    // eslint-disable-next-line react/prop-types
+    props['data-length'] && M.CharacterCounter.init(inputRef);
+  }, [props]);
+
+  useEffect(() => {
+    M.updateTextFields();
+  }, [value]);
+
+  const sizes = { s, m, l, xl };
+
+  let responsiveClasses;
+  if (!noLayout) {
+    responsiveClasses = { col: true };
+    constants.SIZES.forEach(size => {
+      responsiveClasses[size + sizes[size]] = sizes[size];
+    });
   }
 
-  componentDidMount() {
-    if (typeof M !== undefined) {
-      // eslint-disable-next-line react/prop-types
-      this.props['data-length'] && M.CharacterCounter.init(this.inputRef);
-    }
+  const wrapperClasses = cx('input-field', responsiveClasses);
+
+  let computedType;
+  if (type) {
+    computedType = type;
+  } else if (password) {
+    computedType = 'password';
+  } else if (email) {
+    computedType = 'email';
+  } else {
+    computedType = 'text';
   }
 
-  componentDidUpdate(prevProps) {
-    const { value } = this.props;
+  const inputProps = {
+    placeholder,
+    type: computedType,
+    id: getNnewId(),
+    value: newValue,
+    defaultValue,
+    disabled,
+    ...props
+  };
+
+  const renderLabel = () =>
+    label && (
+      <label
+        className={cx({
+          active: newValue || placeholder,
+          'label-icon': typeof label !== 'string'
+        })}
+        data-success={success}
+        data-error={error}
+        htmlFor={inputProps.id}
+      >
+        {label}
+      </label>
+    );
 
-    if (value !== prevProps.value && typeof M !== 'undefined') {
-      M.updateTextFields();
-    }
-  }
+  const renderHelper = () =>
+    (error || success) && (
+      <span className="helper-text" data-error={error} data-success={success} />
+    );
 
-  render() {
-    const {
-      children,
-      s,
-      m,
-      l,
-      xl,
-      disabled,
-      noLayout,
-      placeholder,
-      icon,
-      label,
-      inputClassName,
-      success,
-      error,
-      password,
-      email,
-      validate,
-      defaultValue,
-      value,
-      type,
-      ...other
-    } = this.props;
-
-    const sizes = { s, m, l, xl };
-
-    let responsiveClasses;
-    if (!noLayout) {
-      responsiveClasses = { col: true };
-      constants.SIZES.forEach(size => {
-        responsiveClasses[size + sizes[size]] = sizes[size];
-      });
-    }
+  const renderIcon = () => {
+    if (!icon) return;
 
-    const wrapperClasses = cx('input-field', responsiveClasses);
-
-    let computedType;
-    if (type) {
-      computedType = type;
-    } else if (password) {
-      computedType = 'password';
-    } else if (email) {
-      computedType = 'email';
-    } else {
-      computedType = 'text';
+    if (typeof icon === 'string') {
+      return <i className="material-icons prefix">{icon}</i>;
```

Comment:
as per other components, we only accept `icon` as a node now, but you still need to append prefix: `return React.cloneElement(icon, { className: 'prefix' });`